.upstream-tests/test/cuda/proclaim_return_type.pass.cpp

```diff
@@ -0,0 +1,65 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++98, c++03
```

Comment:
@wmaxey I'm using `__invoke` which is used in `result_of` which is supported in C++11. I was hoping that it's fine to rely on `__invoke`. Please, let me know if I'm missing something. 

I also found that this test fails in GCC 4.8.5. This happens due to some internal compiler bug (here's a [reproducer](https://godbolt.org/z/5EMPWheKM)). I decided to mark this test as unsupported for GCC 4. I hope this is fine, since libcu++ supports GCC 5+.

Reply:
We typically don't need to worry about anything pre-C++11 or GCC 4. That is fine. :)